index.js

```diff
@@ -7,11 +7,11 @@ function escapeCarriageReturn(txt) {
   if (!txt) return "";
   if (!/\r/.test(txt)) return txt;
   txt = txt.replace(/\r+\n/gm, "\n"); // \r followed by \n --> newline
-  while (/\r[^$]/.test(txt)) {
-    var base = /^(.*)\r+/m.exec(txt)[1];
-    var insert = /\r+(.*)$/m.exec(txt)[1];
-    insert = insert + base.slice(insert.length, base.length);
-    txt = txt.replace(/\r+.*$/m, "\r").replace(/^.*\r/m, insert);
+  while (/\r/.test(txt)) {
+    txt = txt.replace(
+        /^([^\r\n]*)\r+([^\r\n]*)/mg,
+        (_, base, insert) => insert + base.slice(insert.length),
+    );
```

Comment:
Can we change this to a normal function?
We didn't use arrow functions before so this might break some users with ancient Javascript runtime versions. Not sure how relevant this is nowadays though but it's an easy change from our side
```suggestion
    txt = txt.replace(/^([^\r\n]*)\r+([^\r\n]*)/gm, function (_, base, insert) {
      return insert + base.slice(insert.length);
    });
```